### PR TITLE
fix: handle arrow function without curly braces body

### LINKED ISSUE #57: Generation fails when API function is defined as arrow function expression
When API function is defined as arrow function expression:

```
/**
 * @pact
 * @pact-method GET
 * @pact-path /example
 */
const getExample = () => axios.get('/example')
```

pact-gen-ts fails with following error:

```
> pact-gen-ts

InvalidOperationError: A child of the kind Block was expected.
    at Object.throwIfNullOrUndefined (C:\Users\hlukafis\projects\payments-in-ui\node_modules\@ts-morph\common\dist\ts-morph-common.js:471:19)
    at ArrowFunction.getFirstChildByKindOrThrow (C:\Users\hlukafis\projects\payments-in-ui\node_modules\ts-morph\dist\ts-morph.js:3734:30)
    at InteractionCreator.getResponseBodyForApiFunction (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\interaction-creator.js:73:50)
    at InteractionCreator.getInteractionForPactJsDoc (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\interaction-creator.js:57:43)
    at Array.map (<anonymous>)
    at InteractionCreator.findAllInteractions (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\interaction-creator.js:113:31)
    at Function.getAllInteractionsInFile (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\interaction-creator.js:109:61)
    at readInteractionsFromFiles (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\create-pacts.js:51:71)
    at createPactForProvider (C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\create-pacts.js:38:35)
    at C:\Users\hlukafis\projects\payments-in-ui\node_modules\pact-gen-ts\lib\core\create-pacts.js:28:61
```

### PR DESCRIPTION
**Pull request by Fiszcz**
Resolves #57

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1213,7 +1213,7 @@
         },
         "node_modules/@ts-morph/common": {
             "version": "0.12.2",
-            "resolved": "https://repo.tech.hl.uk/repository/npm/@ts-morph/common/-/common-0.12.2.tgz",
```
Comment (Fiszcz) [inline package-lock.json:1216]: xD @crpc Remember that not every developer works at HL.

File: src/core/interaction-creator.ts [hunk 2]
```diff
@@ -78,9 +78,9 @@ export class InteractionCreator {
     };
 
     private getResponseBodyForApiFunction = (apiFunctionNode: tsMorph.Node) => {
-        const functionBody = apiFunctionNode.getFirstChildByKindOrThrow(ts.SyntaxKind.Block);
+        const functionBody = apiFunctionNode.getFirstChildByKind(ts.SyntaxKind.Block);
         const responseBodyType =
-            InteractionCreator.getResponseTypeFromFunctionBody(functionBody) ||
+            (functionBody && InteractionCreator.getResponseTypeFromFunctionBody(functionBody)) ||
```
Comment (pawfa) [inline src/core/interaction-creator.ts:83]: What do you think about making this function a bit more descriptive - for me it is not easy to follow the flow - maybe a bit more code showing exactly what cases there are might be useful in understanding this later:
```
    private getResponseBodyForApiFunction = (apiFunctionNode: tsMorph.Node) => {
        const functionBody = apiFunctionNode.getFirstChildByKind(ts.SyntaxKind.Block);
        const responseBodyForFunctionWithBlockType = functionBody && InteractionCreator.getResponseTypeFromFunctionBody(functionBody);
        
        if(responseBodyForFunctionWithBlockType) {
            const basicTypeRepresentationOfResponse = getTypeRepresentation(responseBodyForFunctionWithBlockType, this.sourceFile, true);
            return {
                body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
                matchingRules: changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
            };
        }
        
        const responseBodyReturnFunctionType = InteractionCreator.getReturnTypeOfFunction(apiFunctionNode.getType());
        const basicTypeRepresentationOfResponse = getTypeRepresentation(responseBodyReturnFunctionType, this.sourceFile, true);
        
        return {
            body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
            matchingRules: changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
        };
    };
```
We have clear distinction that there are 2 main cases: one for function with curly brackets with possible pact-response-body and second for getting type for arrow without curly brackets

**Review by pawfa — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (crpc) [reply package-lock.json:1216]: There's no chance I committed that, I'm clearly being framed here... :D

**Review by crpc — APPROVED** [review]

File: src/core/interaction-creator.ts (hunk 2 above)
Comment (Fiszcz) [reply src/core/interaction-creator.ts:83]: Yeah, I know that that code is not enough clear and obvious and there is a place for improvement.
But I think that your proposition is not also enough solution, there are two repeats and names also are not better than current.

**Comment by lukaszfiszer** [comment]
Thanks! Fixed version is 0.9.1

**Pull request merged**